src/PhpSpec/Console/Application.php

```diff
@@ -57,9 +58,17 @@ public function getContainer()
      */
     public function doRun(InputInterface $input, OutputInterface $output)
     {
+        $helperSet = $this->getHelperSet();
         $this->container->set('console.input', $input);
         $this->container->set('console.output', $output);
-        $this->container->set('console.helper.dialog', $this->getHelperSet()->get('dialog'));
+        $this->container->set('console.helper.dialog', $helperSet->get('dialog'));
```

Comment:
@ciaranmcnulty I suggest adopting the same policy than Behat: considering that people are not meant to instantiate the PhpSpec classes themselves (this is done in the DI container) at least for these levels of the codebase. This means we can change the constructor arguments as we want for refactorings (and actually, you are already relying on this as you added a mandatory constructor argument in the IO).
This means you could move ``askAndValidate`` to the prompter as well